olive/cli/auto_opt.py

```diff
@@ -0,0 +1,265 @@
+            },
+            {
+                "name": "latency",
+                "type": "latency",
+                "sub_types": [
+                    {"name": "avg", "priority": 2, "goal": {"type": "percent-min-improvement", "value": 1}},
+                ],
+                "data_config": "data_config",
+                "user_config": {"io_bind": True},
+            },
+        ]
+    }
+}
+
+TEMPLATE = {
+    "input_model": {"type": "HfModel"},
+    "auto_optimizer_config": {},
+    "search_strategy": {"execution_order": "joint", "search_algorithm": "tpe", "num_samples": 5, "seed": 0},
+    "systems": {
+        "local_system": {
+            "type": "LocalSystem",
+            "accelerators": [{"device": "cpu", "execution_providers": ["CPUExecutionProvider"]}],
+        }
+    },
+    "host": "local_system",
+    "evaluators": EVALUATE_TEMPLATE,
+    "evaluator": "common_evaluator",
+    "target": "local_system",
+}
+
+
+class AutoOptCommand(BaseOliveCLICommand):
+    allow_unknown_args: ClassVar[bool] = True
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "auto-opt",
+            help=("Automatically performance optimize input model"),
+        )
+
+        add_hf_model_options(sub_parser)
+        add_logging_options(sub_parser)
+
+        # model options
+        model_attributes_group = sub_parser.add_argument_group("model attributes options")
+        model_attributes_group.add_argument(
+            "--model_framework",
+            type=str,
+            default="hf",
+            choices=["hf", "onnx", "pytorch"],
+            help="model type, choose from HfModel and ONNXModel",
+        )
+
+        system_group = sub_parser.add_argument_group("system options")
+        system_group.add_argument(
+            "--device",
+            type=str,
+            default=None,
+            choices=["cpu", "gpu", "npu"],
+            # TODO(anyone): add more devices cpu_spr, vpu, intel_myriad
+            help=(
+                "Device to use for optimization, choose from cpu and gpu. If not specified,"
+                " will deduce from the value of execution providers, CPU/VistisAi for cpu and"
+                " CUDA/Tensorrt/Dml for gpu. If both device and execution providers are not specified,"
+                " default to cpu device with CPUExecutionProvider."
+            ),
+        )
+        system_group.add_argument(
+            "--providers",
+            type=str,
+            nargs="*",
+            choices=["CPU", "CUDA", "Tensorrt", "Dml", "VitisAI", "Qnn"],
+            help="List of execution providers to use for optimization",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group(
+            "dataset options, required for some optimization passes like quantization, and evaluation components"
+        )
+        dataset_group.add_argument(
+            "--data_config_path",
```

Comment:
Could we make this a path to HF/Local to the actual data without requiring additional configuration? What does the data need to look like, how would the user generate it?

Reply:
We need to do the same as 'finetune` command and avoid having user provide json/yaml config file.